src/java_binding/src/lib.rs

```diff
@@ -221,22 +222,46 @@ where
     }
 }
 
-pub enum JavaBindingRow {
+pub enum JavaBindingRowKind {
     Keyed(KeyedRow),
     StreamChunk(StreamChunkRow),
 }
+#[derive(Default)]
+pub struct JavaBindingRowCache {
```

Comment:
Better to rename to like `JavaClassMethodCache`. The term `RowCache` looks like deserializing more rows in advance and store the rows in the cache.